AC-1394 Review Glue API backend docs (#1269)
spryker/spryker-docs: docs/scos/dev/glue-api-guides/202204.0/glue-backend-api/how-to-guides/how-to-create-api-applications.md

---
title: How to create API applications
description: This document describes how to create a new API application
last_updated: September 30, 2022
template: howto-guide-template
redirect_from:
  - /docs/scos/dev/glue-api-guides/202204.0/glue-backend-api/how-to-guides/create-api-application.html
---

> New Glue projects can create API applications. This is what you need to do in order to create one.

Author: ```suggestion
New Glue projects can create API applications. This is what you need to do to create one.

To create an API application, the first thing to do is configure Docker.
```

Because the backend and storefront APIs have different sets of configured services, they also have different settings in Docker. Therefore, first of all, ensure your `deploy.yml` file contains the correct setting for `application`. The available options are listed in the following table:

| OPTION | MEANING |
| --- | --- |
| `glue` | Old application value. For the new APIs, choose one of the following options. |

> | `glue-storefront` | Application that has access to key\_value\_store and search.   |

Author: ```suggestion
| `glue-storefront` | Application that has access to key, value, store and search.   |
```

> | `glue-backend` | Application that has access to the following: <ul><li>database</li><li>broker</li><li>key\_value\_store</li><li>session</li><li>search</li></ul> |

Author: ```suggestion
| `glue-backend` | Application that has access to the following: <ul><li>database</li><li>broker</li><li>key, value, store</li><li>session</li><li>search</li></ul> |
```

---
title: How to create API applications
description: This document shows how to create a new API application
last_updated: September 30, 2022
template: howto-guide-template
redirect_from:
  - /docs/scos/dev/glue-api-guides/202204.0/glue-backend-api/how-to-guides/create-api-application.html
---

New Glue projects can create API applications. This is what you need to do in order to create one.

> ### Docker configuration

Author: ```suggestion
```

> First of all, the backend and storefront API have different settings in Docker (they have different sets of services configured for them). Make sure your `deploy.yml` contains the correct setting for `application`. Available options are as follows:

Author: Since => because 
https://developers.google.com/style/word-list#since

    }

}
```
</details>

Each method in `CustomApiApplication` represents a step in the API application request flow. It can be used as an extension point into each of the steps.

This application extends `RequestFlowAwareApiApplication`, which means that this API application follows the default Glue workflow. This is beneficial because it lets you make the most use of the conventions and features in Spryker that wire into the request flow.

> If your API uses its own workflow, you can opt for extending `RequestFlowAgnosticApiApplication`. This kind of application has everything (a separate set of application plugins, boot, and run methods) but not the request flow actions. One of these is the old Glue application. Here is an example of the application: 

Author: We need Ahmed to help us out here. But he is on vacation
